File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest21173.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest21173 {

    @GetMapping("/BenchmarkTest21173")
    public void BenchmarkTest21173(@RequestHeader("Authorization") String authorization, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String authHeader = authorization != null ? authorization : "";
        String data;
        if (authHeader.length() > 256) { data = authHeader.substring(0, 256); }
        else { data = authHeader; }
        String processed = org.owasp.encoder.Encode.forHtml(data);
        response.getWriter().print("<div>" + processed + "</div>");
    }
}

File: Benchmarks/enterprise/python/django/testcode/benchmark_test_21173.py
# SPDX-License-Identifier: Apache-2.0
from django.http import JsonResponse


def BenchmarkTest21173(request):
    forwarded_ip = request.META.get('HTTP_X_FORWARDED_FOR', '')
    int(str(forwarded_ip))
    return JsonResponse({"saved": True})
